aquadoggo/migrations/20220617115933_create-tasks.sql

```diff
@@ -0,0 +1,8 @@
+-- SPDX-License-Identifier: AGPL-3.0-or-later
+
+CREATE TABLE IF NOT EXISTS tasks (
+    name              TEXT      NOT NULL,
+    document_id       TEXT      NULL,
+    document_view_id  TEXT      NULL,
+    PRIMARY KEY (name, document_id, document_view_id)
```

Comment:
I was just stumbling about this primary key with `null` columns. If I understand correctly this is actually only possible in Sqlite:

> According to the SQL standard, PRIMARY KEY should always imply NOT NULL. Unfortunately, due to a bug in some early versions, this is not the case in SQLite. Unless the column is an INTEGER PRIMARY KEY or the table is a WITHOUT ROWID table or a STRICT table or the column is declared NOT NULL, SQLite allows NULL values in a PRIMARY KEY column. SQLite could be fixed to conform to the standard, but doing so might break legacy applications. Hence, it has been decided to merely document the fact that SQLite allows NULLs in most PRIMARY KEY columns. 

https://www.sqlite.org/lang_createtable.html